google/cloud/internal/curl_rest_client.cc

```diff
@@ -62,9 +60,8 @@ Status MakeRequestWithPayload(
   for (auto const& p : payload) {
     content_length += p.size();
   }
-  if (content_length > 0) {
-    impl.SetHeader(absl::StrCat("content-length: ", content_length));
-  }
+
+  impl.SetHeader(absl::StrCat("Content-Length: ", content_length));
```

Comment:
Please make sure we test both with zero and non-zero values for `content_length`.